internal/types/types.go

```diff
@@ -189,3 +192,58 @@ func deepCopy(value any) any {
 		panic(fmt.Sprintf("types.deepCopy: unsupported type: %[1]T (%[1]v)", value))
 	}
 }
+
+// JSONSyntax formats a FerretDB value as it's JSON counterpart. This is useful when embedding an invalid value in an
+// error.
+// For example, consider a query:
+// 	db.values.find( { *someFilter* }, { arr: { $slice: { a: { b: 3 }, b: [ 1, 2 ], x: 3 } } } )
+//
+// This query returns an error that starts with:
+// 	"Invalid $slice syntax. The given syntax { $slice: { a: { b: 3 }, b: [ 1, 2 ], x: 3 } } did not match the find()
+// 	syntax because :: /.../"
+//
+// JSONSyntax makes sure that the value in the { $slice: /.../ } clause gets formatted correctly.
+// It differs from fjson package in its purpose.
+// JSONSyntax is needed to match the output of MongoDB in the visually particular case of embedding a
+// provided value into an error message. In contrast, fjson package aims to format values according to FJSON standard.
+func JSONSyntax(value any) string {
```

Comment:
Ono more thing (sorry didn't say it before)
Please see https://pkg.go.dev/github.com/FerretDB/FerretDB/internal/types#hdr-Mapping and note the order of the types ( summarised `Composite types` and `Scalar types`. 
And see that in the `fjson` package in the switch of func `fromFJSON` types are ordered like in the doc.
It helps not to omit any type. 

Could you please reorder types in this switch to the standard?